Cargo.toml

```diff
@@ -80,6 +80,266 @@ resolver = "2"
+mz-ssh-util = { path = "src/ssh-util" }
+mz-stash = { path = "src/stash" }
+mz-stash-debug = { path = "src/stash-debug" }
+mz-storage = { path = "src/storage" }
+mz-storage-client = { path = "src/storage-client" }
+mz-testdrive = { path = "src/testdrive" }
+mz-timely-util = { path = "src/timely-util" }
+mz-transform = { path = "src/transform" }
+mz-walkabout = { path = "src/walkabout" }
+native-tls = "0.2.11"
+nix = "0.26.1"
+num = "0.4.0"
+num_cpus = "1.14.0"
+num_enum = "0.5.7"
+num-derive = "0.3"
+num-traits = "0.2.15"
+once_cell = "1.16.0"
+open = "3.2.0"
+openssh = { version = "0.9.8", default-features = false }
+openssh-mux-client = "0.15.5"
+openssl = "0.10.48"
+openssl-probe = "0.1.2"
+openssl-sys = "0.9.80"
+opentelemetry = { git = "https://github.com/MaterializeInc/opentelemetry-rust.git" }
+opentelemetry-otlp = { git = "https://github.com/MaterializeInc/opentelemetry-rust.git" }
+ordered-float = "3.4.0"
+os_info = "3.5.1"
+parquet2 = { version = "0.17.1", default-features = false }
+paste = "1.0.11"
+phf = "0.11.1"
+phf_codegen = "0.11.1"
+pin-project = "1.0.12"
+polonius-the-crab = "0.3.1"
+postgres = { git = "https://github.com/MaterializeInc/rust-postgres" }
+postgres_array = { git = "https://github.com/MaterializeInc/rust-postgres-array" }
+postgres-openssl = { git = "https://github.com/MaterializeInc/rust-postgres" }
+postgres-protocol = { git = "https://github.com/MaterializeInc/rust-postgres" }
+pprof = "0.11.1"
+predicates = "2.1.4"
+proc-macro2 = "1.0.47"
+prometheus = { version = "0.13.3", default-features = false }
+proptest = { git = "https://github.com/MaterializeInc/proptest.git", default-features = false, features = ["std"] }
+proptest-derive = { git = "https://github.com/MaterializeInc/proptest.git" }
+prost = "0.11.9"
+prost-build = "0.11.9"
+prost-reflect = "0.9.2"
+protobuf-native = "0.2.1"
+protobuf-src = "1.1.0"
+quote = "1.0.23"
+rand = "0.8.5"
+rdkafka = { git = "https://github.com/MaterializeInc/rust-rdkafka.git" }
+rdkafka-sys = { git = "https://github.com/MaterializeInc/rust-rdkafka.git" }
+regex = "1.7.0"
+regex-syntax = "0.6.28"
+reqwest = "0.11.13"
+rlimit = "0.8.3"
+rocksdb = { version = "0.21.0", default-features = false }
+rpassword = "7.2.0"
+ryu = "1.0.12"
+schemars = "0.8.11"
+scopeguard = "1.1.0"
+security-framework = "2.7.0"
+segment = { version = "0.2.1", default-features = false }
+semver = "1.0.16"
+sentry = "0.29.1"
+sentry-tracing = "0.29.1"
+serde = "1.0.152"
+serde_json = "1.0.89"
+serde_plain = "1.0.1"
+serde_regex = "1.1.0"
+serde_urlencoded = "0.7.1"
+sha1 = "0.10.5"
+sha2 = "0.10.6"
+shell-words = "1.1.0"
+similar = "2.2.1"
+smallvec = "1.10.0"
+snap = "1.1.0"
+socket2 = "0.4.7"
+ssh-key = "0.4.3"
+stacker = "0.1.15"
+syn = "1.0.107"
+sysctl = "0.5.4"
+sysinfo = "0.27.2"
+tar = "0.4.38"
+tempfile = "3.3.0"
+termcolor = "1.1.3"
+thiserror = "1.0.37"
+tiberius = { version = "0.11.3", default-features = false }
+tikv-jemalloc-ctl = "0.5.0"
+tikv-jemallocator = "0.5.0"
```

Comment:
These are optional dependencies. It would be great if we could keep it this way because several memory profiling tools have a hard time with jemalloc and work much better with the system allocator.

Reply:
(This is also why the lint fails)

Reply:
Huh, I'm surprised this doesn't work as is. AFAICT the dependencies are still optional in the package itself; it's just that if they _are_ enabled they're meant to use the version here.

Reply:
Totally agree that we should keep these as optional deps, it was through investigating this lint that I found default-features don't really work with workspace dependencies. I filed https://github.com/rust-lang/cargo/issues/12162 to discuss this

Reply:
The described behavior seems workable for us. Seems like it'll be annoying for the author of this PR (:D), but not too annoying to deal with once it's in place. There are very few crates where we want `default-features = false` in some places but not others; usually every Materialize crates wants to turn of the default features or none.